src/pudl/helpers.py

```diff
@@ -339,6 +339,227 @@ def is_doi(doi):
     return bool(re.match(doi_regex, doi))
 
 
+def full_timeseries_temporal_merge(
+    left: pd.DataFrame,
+    right: pd.DataFrame,
+    left_date_col: str = "report_date",
+    right_date_col: str = "report_date",
+    new_date_col: str = "report_date",
+    shared_merge_cols: List[str] = [],
+    temporal_merge_cols: Literal["year", "quarter", "month", "day"] = "year",
+    merge_type: str = "inner",
+    report_at_start=True,
+    start: str = None,
+    end: str = None,
+    freq: Literal["AS", "QS", "MS", "D"] = "MS",
+    fill=True,
+    **kwargs,
+):
+    """
+    Merge dataframes with different date frequencies and expand to a full timeseries.
+
+    Arguments: see arguments for ``mexed_temporal_gran_merge`` and ``expand_timeseries``
+    """
+    out = mixed_temporal_gran_merge(
+        left,
+        right,
+        left_date_col,
+        right_date_col,
+        shared_merge_cols,
+        temporal_merge_cols,
+        merge_type,
+        report_at_start,
+        **kwargs,
+    )
+    out = expand_timeseries(
+        df=out,
+        date_col=new_date_col,
+        start=start,
+        end=end,
+        freq=freq,
+        id_cols=shared_merge_cols,
+        fill=fill,
+    )
+    return out
+
+
+def mixed_temporal_gran_merge(
+    left: pd.DataFrame,
+    right: pd.DataFrame,
+    left_date_col: str = "report_date",
+    right_date_col: str = "report_date",
+    new_date_col: str = "report_date",
+    shared_merge_cols: List[str] = [],
+    temporal_merge_cols: List[str] = ["year"],
```

Comment:
Can you constrain the values that can be in a list, e.g. "year", "month", "quarter", "day"? It's a list not a string so `Literal` won't work?